### ISSUE TITLE
GPU usage goes to 100% if the game window isnt minimize, but on minimise goes to 10-30%

### ISSUE THREAD

**Issue by danielkariv** [BastiaanOlij/godot-vr-weapons#1]
So, as the title says, everytime i try to debug the game, the GPU usage goes to 100% just because the window exist. if I minimize it, the GPU usage goes down a lot. It even hurt the VR rendering, from 45fps and 100% to 90% and 10%. 
I tried to:
- remove meshes and effects - didn't help.
- disabled the viewport for the window - didn't help. 

I don't know what else I can try, I sitting on it over a day or so, at first i thought it is a Godot performance issues, till I figure out the minimize trick that make it way down.

System:
I7 4770
32GB RAM
GTX 1070 ( with the latest update)
Project and engine on a SSD.
System runs on Windows 10 64bit.

**Comment by BastiaanOlij** [comment]
Hey @danielkariv ,

What a weird issue, I haven't noticed that before. It almost sounds like the v-sync on the headset isn't throttling things (the v-sync on the desktop is turned off to prevent throttling to 60fps). Are you running Godot in multi threading mode?

Your setup is nearly identical to mine. I'll have to do some experimenting over the weekend and see what result I get.

What headset are you using?

**Comment by danielkariv** [comment]
Thread model for rendering is set to Multi-Threaded but i just tried to change it to single thread (safe and unsafe), the issue still exist. V-Sync is turn off, both by the script and by the project settings.
The headset is Oculus Rift CV1.

**Comment by BastiaanOlij** [comment]
Strange, I have the same headset. 

Mind you, I have noticed that output to screen does slow things down which is why I tent to keep the window small. No idea why though as it's doing nothing more then blitting the buffer to screen but nothing as extreme as you're suggesting.
Maybe its having trouble because it ends up doing that at 90fps. I'm going to have to ask Reduz about that, see if we can come up with a better idea to handle this.

**Comment by danielkariv** [comment]
Okay so i tested it more in the last 'few' hours. 
So first i tried to disable HDR, thought maybe it will change something, it didnt ( well expect maybe 5% difference but it is totally in the margin of error).
Next it tried to disable ARVR option in the script. Now the GPU is around 50%.
![1](https://user-images.githubusercontent.com/38776931/57562477-ee7ff880-739b-11e9-93fd-4bfeb25233d3.png)
and this is when the ARVR is ON:
![2](https://user-images.githubusercontent.com/38776931/57562485-f6d83380-739b-11e9-9525-1d0dd6f906bb.png)
Okay so because of that, i thought maybe it is driver issue with SteamVR and Oculus, so i deleted the OpenVR addon, and tried to run the project as a normal game ( with ofcourse, same res as the Rift, and 90 FPS), and it again showed 45%.
![3](https://user-images.githubusercontent.com/38776931/57562486-f93a8d80-739b-11e9-99fa-cd82c6fb2c8b.png)

More than that, if you force the game to run on lower FPS ( like 45, or lower) the GPU usage get lower too.
Also about the minimize trick, I found out when the game isnt focus, it doesnt render it anymore, meaning that 100% -> 10% was basically the game not rendering anything.

So basically, I don't think it is a addon issue but Godot render engine. I guess rendering 90 frames, for each screen on the device is very hard. But just for be sure, can you test it on your machine to see if you too reach to very high GPU usage?

**Comment by BastiaanOlij** [comment]
Hi @danielkariv ,

When you turn ARVR off you're turning of stereoscopic rendering and outputting to the HMD, that is a large chunk of the processing the GPU needs to do taken away. 

The thing is, when you're just rendering to the desktop, you are rendering 1920x1080 or something like that at 60fps. 
When you are rendering stereoscopic for your Oculus headset you are rendering 2160x2400 x 2 at 90fps and output a copy to your display, your GPU has to do roughly 8x more work. 

Seeing we currently use a GI probe for lighting in this example things start to really take a tax. Dealing with that is a subject for a future tutorial

You can mitigate that a little in your openvr settings, you can lower the default oversampling to say 1.5 or 1.25 which will render each eye to a smaller resolution viewport. For VR you need to render at much higher resolutions then the headset supports because the lens distortion will cause compression of the final image.
